Lib/multiprocessing/shared_memory.py

```diff
@@ -134,7 +243,7 @@ def __init__(self, *args, buffer=None, **kwargs):
                 assert hasattr(self, "_proxied_type")
             try:
                 existing_type.__init__(self, *args, **kwargs)
-            except:
+            except Exception:
```

Comment:
Agreed this deserves a comment.  Anecdotally (but unfortunately I do not immediately remember which class), one or more classes that I successfully tested this function on had enforced restrictions on what keyword arguments were allowed to be passed into its `__init__` yet extra keyword arguments were permissible/ignored when passed into its `__new__`; for this class, the filling of values into a new object instance was accomplished by its `__new__` and the `__init__` was superfluous for the purposes of the `shareable_wrap` function.  In order to add a comment, I need to find this in my notes but before doing so I would like to advocate for the removal of the `shareable_wrap` function from the module.

As described in the docstring for `shareable_wrap`, its purpose was to make it quick and easy to put objects supporting the buffer protocol as both producer and consumer onto shared memory:
```
>>> import numpy as np
>>> a = np.array([4, 5, 6, 7])
>>> print(a)
[4 5 6 7]
>>> from multiprocessing import shared_memory
>>> shm_a = shared_memory.shareable_wrap(a)
>>> shm_a
ndarrayShareable(shape=(4,), strides=(8,), dtype=dtype('int64'), shmem_name='psm_25371_85606', cls=<class 'numpy.ndarray'>)
>>> print(shm_a)
[4 5 6 7]
>>> import pickle
>>> with open("./arr.pkl", "wb") as f:
...     pickle.dump(shm_a, f)
... 
>>> with open("./arr.pkl", "rb") as f:
...     shm_b = pickle.load(f)  # Load in a different process
... 
>>> shm_b  # Note the same shared memory block name
ndarrayShareable(shape=(4,), strides=(8,), dtype=dtype('int64'), shmem_name='psm_25371_85606', cls=<class 'numpy.ndarray'>)
>>> shm_b[-1] = 999
>>> print(shm_a)
[  4   5   6 999]
>>> len(pickle.dumps(shm_a))  # Note the size of the serialized representation of a small shared numpy array
122
>>> c = np.array(range(50000))
>>> shm_c = shared_memory.shareable_wrap(c)
>>> len(shm_c)
50000
>>> len(pickle.dumps(shm_c))  # Note the size of the much larger shared array's serialized representation
123
>>> len(pickle.dumps(c))
400162
```

I propose that `shareable_wrap` should be removed from the code for three reasons:
(1) We have no examples of data types in the core of Python that are __both__ producers and consumers of the buffer protocol, hence I do not have a way to demonstrate its use without referring to 3rd party library objects like `numpy.ndarray`.
(2) Feedback from others who have tested `shareable_wrap` generally found it easy to use but found the representation (the output from `__repr__`) of the wrapped objects to be surprising. I am unsure of how to solve this sense of surprise in a truly generic way.
(3) The buffer protocol makes no specification at the Python object api level for how to supply the required inputs.  This `shareable_wrap` function is a pure Python function.  As such it took many iterations to adapt this function to handle the widest possible variety of 3rd party library objects, but there are undoubtedly scenarios / new kinds of objects which it will not be able to handle.

Reply:
I included this function to provoke discussion around its limitations while demonstrating some of what is possible.  This is also why I skipped including it in the documentation for now.

Reply:
The consensus seemed to favor removing it.
I have removed it in commit 885592b56f555c03e6444c7de1839c38c4792fb6.